### PR TITLE
Add --compile-all to sky_snapshot

### PR DESCRIPTION
**Pull request by johnmccutchan**
@tvolkert

### CODE REVIEW

**Comment by tvolkert** [comment]
LGTM

File: snapshotter/main.cc [hunk 1]
```diff
@@ -206,6 +208,28 @@ int CreateSnapshot(const ftl::CommandLine& command_line) {
...
+              << Dart_GetError(load_result)
+              << std::endl;
+    return 1;
+  }
+
+  const bool compile_all_mode = command_line.HasOption(kCompileAll, nullptr);
+  if (compile_all_mode) {
+    // Compile all the code loaded into the isolate. This will eagerly detect
+    // syntax errors.
+    Dart_Handle compile_all_results = Dart_CompileAll();
```
Comment (abarth) [inline snapshotter/main.cc]: Will this change the generated snapshot file?

**Comment by johnmccutchan** [comment]
@abarth no

**Comment by johnmccutchan** [comment]
@abarth Actually I was mistaken but I have a follow up CL which won't affect the output.

**Pull request merged**